### ISSUE TITLE
(arm64) Add documentation for arm based mac environments (flutter_rust_bridge_codegen)

### ISSUE THREAD

**Issue by mherrerarendon** [fzyzcjy/flutter_rust_bridge#395]
First of all, thank you for writing this library. I find it very useful.

**Is your feature request related to a problem? Please describe.**
The issue I encountered is reproducible on an arm based mac, and is mainly caused by the dart ffigen not having an arm64 architecture support yet. The workaround is to point ffigen to x86_64 llvm libraries instead. However, `flutter_rust_bridge_codegen` by default points to the arm64 llvm libraries (which is strange because the code includes the x86 llvm path [here](https://github.com/fzyzcjy/flutter_rust_bridge/blob/cc909499c3d70e23f2b0ad1e855669524b02958f/frb_codegen/src/config.rs#L138)).
This is the error I was encountering
```
[2022-03-26T19:00:53Z WARN  flutter_rust_bridge_codegen::commands] command="sh" "-c" "dart pub global run ffigen --config \"/var/folders/mq/86hq1ydx6plf4qp7jxlp8qkh0000gn/T/.tmpOMq4GZ\"" stdout=Running in Directory: '/Users/mherrerarendon/Developer/git/sound_test'
     stderr=Unhandled exception:
    Invalid argument(s): Failed to load dynamic library '/opt/homebrew/opt/llvm/lib/libclang.dylib': dlopen(/opt/homebrew/opt/llvm/lib/libclang.dylib, 0x0001): tried: '/opt/homebrew/opt/llvm/lib/libclang.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/lib/libclang.dylib' (no such file), '/opt/homebrew/Cellar/llvm/13.0.1_1/lib/libclang.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/lib/libclang.dylib' (no such file)
```
In other words, it's looking for llvm here `/opt/homebrew/opt/llvm` which houses the arm64 binaries, but not here `/usr/local/homebrew/opt/llvm`, which houses the x86_64 binaries

**Describe the solution you'd like**
Ideally, `flutter_rust_bridge_codegen` would configure ffigen to point to the correct llvm binaries out of the box.
Another solution would be to add temporary documentation around using the `flutter_rust_bridge_codegen` tool on m1 macs

**Comment by fzyzcjy** [comment]
Seems that the codegen has a option to specify the llvm you want, so maybe you can try it.

In addition, if that does not satisfy your needs, maybe make a PR for it - this part (calling llvm) is very simple and easy to understand. I can point you to the file if needed.

**Comment by mherrerarendon** [comment]
Yes, sorry. I forgot to say that adding the `--llvm-path` argument in the code generator does solve the build issue. It did take me a while to understand that that's what I needed to do. My main driver for writing up this issue was to avoid others needing to go through the same discovery process.

As I was looking for a good place to document troubleshooting, I found that this exact issue is already discussed [here](https://github.com/fzyzcjy/flutter_rust_bridge/blob/master/book/src/troubleshooting.md#on-m1-macos--failed-to-load-dynamic-library-opthomebrewoptllvmliblibclangdylib), so maybe we should just close this.

**Comment by fzyzcjy** [comment]
Happy to see you solved the problem :)
